retriever/lib/engine.py

```diff
@@ -680,17 +680,22 @@ def table_name(self, name=None, dbname=None):
                 dbname = ''
         return self.opts["table_name"].format(db=dbname, table=name)
 
-    def to_csv(self):
+    def to_csv(self, db=None):
         # Due to Cyclic imports we can not move this import to the top
         from retriever.lib.engine_tools import sort_csv
-        for _ in list(self.script.urls.keys()):
-            table_name = self.table_name()
+        for table in list(self.script.urls.keys()):
+            table_name = "{}_{}".format(self.database_name(), table)
+            if db == "postgres" or db == "mysql":
```

Comment:
One thing that you should note, `class Engine` is the main class for all other engines and it should be generic as possible.
Each engine `(postgres, mysql, csv and etc)` inherits these methods/functions and if they differ from the general method, that engine overrides(customizes) that method.

Since engines `postgres and mysql`  want to implement something different from the main function `def to_csv()` in the main class, these engines classes `(postgres and mysql)` should instead override the entire method.

In this way we avoid checks for specific variables that do not apply to the general object. ` if db == "postgres" or db == "mysql":`

Reply:
Okay 👍